src/Maestro/Maestro.Web.Tests/SubscriptionsController20200220Tests.cs

```diff
@@ -0,0 +1,485 @@
+                createdSubscription1.SourceRepository.Should().Be(defaultGitHubSourceRepo);
+                createdSubscription1.TargetRepository.Should().Be(defaultGitHubTargetRepo);
+            }
+
+            Api.v2018_07_16.Models.SubscriptionData subscription2 = new Api.v2018_07_16.Models.SubscriptionData()
+            {
+                ChannelName = testChannelName,
+                Enabled = false,
+                SourceRepository = defaultAzdoSourceRepo,
+                TargetRepository = defaultAzdoTargetRepo,
+                Policy = new Api.v2018_07_16.Models.SubscriptionPolicy() { Batchable = false, UpdateFrequency = Api.v2018_07_16.Models.UpdateFrequency.None },
+                TargetBranch = defaultBranchName
+            };
+
+            Subscription createdSubscription2;
+            {
+                IActionResult result = await data.SubscriptionsController.Create(subscription2);
+                result.Should().BeAssignableTo<ObjectResult>();
+                var objResult = (ObjectResult) result;
+                objResult.StatusCode.Should().Be((int) HttpStatusCode.Created);
+                objResult.Value.Should().BeAssignableTo<Subscription>();
+                createdSubscription2 = (Subscription) objResult.Value;
+                createdSubscription2.Channel.Name.Should().Be(testChannelName);
+                createdSubscription2.Policy.Batchable.Should().Be(false);
+                createdSubscription2.Policy.UpdateFrequency.Should().Be(Api.v2018_07_16.Models.UpdateFrequency.None);
+                createdSubscription2.TargetBranch.Should().Be(defaultBranchName);
+                createdSubscription2.SourceRepository.Should().Be(defaultAzdoSourceRepo);
+                createdSubscription2.TargetRepository.Should().Be(defaultAzdoTargetRepo);
+            }
+
+            // List all (both) subscriptions, spot check that we got both
+            {
+                IActionResult result = data.SubscriptionsController.ListSubscriptions();
+                result.Should().BeAssignableTo<ObjectResult>();
+                var objResult = (ObjectResult) result;
+                objResult.StatusCode.Should().Be((int) HttpStatusCode.OK);
+                List<Subscription> listedSubs = ((IEnumerable<Subscription>) objResult.Value).ToList();
+                listedSubs.Count.Should().Be(2);
+                listedSubs[0].Enabled.Should().Be(true);
+                listedSubs[0].TargetRepository.Should().Be(defaultGitHubTargetRepo);
+                listedSubs[1].Enabled.Should().Be(false);
+                listedSubs[1].TargetRepository.Should().Be(defaultAzdoTargetRepo);
+            }
+            // Use ListSubscriptions() params at least superficially to go down those codepaths
+            {
+                IActionResult result = data.SubscriptionsController.ListSubscriptions(defaultAzdoSourceRepo, defaultAzdoTargetRepo, createdSubscription2.Channel.Id, false);
```

Comment:
It slightly feels like this should be a separate test (one called "ListAllNoParameters", one called "ListSomeWithParameters"), but that's a lot of setup.

Reply:
agreed, left as-is.  Yeah it's a big fact but it's easy to debug.